lib/nuget-parser/dotnet-core-parser.js

```diff
@@ -123,38 +121,43 @@ function getTargetObjToRun(manifest) {
   return manifest.targets[selectedTargetKey];
 }
 
+module.exports = {
+  parse: function (tree, fileContent, fileContentParser) {
+    return new Promise(function parseFileContents(resolve) {
+      debug('Trying to parse dot-net-cli manifest');
+      const manifest = fileContentParser.parse(fileContent);
 
-function parse(fileContent, packageTree) {
-  const manifest = JSON.parse(fileContent);
-
-  if (!manifest.project) {
-    throw new Error('Project field was not found in project.assets.json');
-  }
-
-  if (manifest.project.version) {
-    packageTree.version = manifest.project.version;
-  }
+      if (!manifest.project) {
+        throw new Error('Project field was not found in project.assets.json');
+      }
 
-  const selectedFrameworkObj = getFrameworkObjToRun(manifest);
-  const selectedTargetObj = getTargetObjToRun(manifest);
+      if (manifest.project.version) {
+        tree.version = manifest.project.version;
+      }
 
-  initFreqDepsDict();
+      tree.meta.targetFramework = getFrameworkToRun(manifest);
+      const selectedFrameworkObj = manifest.project.frameworks[tree.meta.targetFramework];
+      const selectedTargetObj = getTargetObjToRun(manifest);
 
-  const directDependencies = collectFlatList(selectedFrameworkObj.dependencies);
-  debug(`directDependencies: '${directDependencies}'`);
+      initFreqDepsDict();
 
-  directDependencies.forEach(function (directDep) {
-    debug(`First order dep: '${directDep}'`);
-    buildTreeRecursive(selectedTargetObj, directDep, packageTree, 0);
-  });
+      const directDependencies = collectFlatList(selectedFrameworkObj.dependencies);
+      debug(`directDependencies: '${directDependencies}'`);
 
-  if (!_.isEmpty(freqDeps.dependencies.dependencies)) {
-    packageTree.dependencies['freqSystemDependencies'] = freqDeps.dependencies;
-  }
-  // to disconnect the object references inside the tree
-  // JSON parse/stringify is used
-  let pathedTree = JSON.parse(JSON.stringify(packageTree.dependencies));
-  packageTree.dependencies = pathedTree;
-}
+      directDependencies.forEach(function (directDep) {
+        debug(`First order dep: '${directDep}'`);
+        buildTreeRecursive(selectedTargetObj, directDep, tree, 0);
+      });
 
-module.exports = parse;
+      if (!_.isEmpty(freqDeps.dependencies.dependencies)) {
+        tree.dependencies['freqSystemDependencies'] = freqDeps.dependencies;
+      }
+      // to disconnect the object references inside the tree
+      // JSON parse/stringify is used
+      tree.dependencies = JSON.parse(JSON.stringify(tree.dependencies));
+      resolve(tree);
+    }).catch(function (err) {
+      throw (err);
```

Comment:
This doesn't make sense. Just delete `catch ` block.

Reply:
deleted